### PR TITLE
26 report ns mismatch

### PR DESCRIPTION
**Pull request by wtoorop**
Send DNS Error report on NS set mismatch

### CODE REVIEW

File: draft-ietf-dnsop-ns-revalidation.mkd [hunk 1]
```diff
@@ -175,31 +177,33 @@ Attacks exploiting lack of this revalidation have been described in {{GHOST1}},
...
 
--   The names from the NS RRset from a validating NS query may differ from the names in NS RRset in the referral response.
-    Outstanding validation queries for "glue" that do not match names in the authoritative NS RRset be discarded, or they may be left running to completion.
-    Their result will no longer be used in queries for the zone.
-    Outstanding validation queries for "glue" that do match names in the authoritative NS RRset must be left running to completion.
-    They do not need to be re-queried after reception of the authoritative NS RRset (see [](#upgrade-addresses)).
+Additional validation queries for the "glue" resource records of referral responses (if not already authoritatively present in cache) may be sent with the validation query for the NS RRset as well.
+Positive responses will be cached authoritatively and replace the non authoritative "glue" entries.
+Successive queries directed to the same zone will be directed to the authoritative values for the names of the NS RRset in the referral response.
```
Comment (vixie) [inline draft-ietf-dnsop-ns-revalidation.mkd]: i think "to the authoritative nameservers denoted in" sounds better than "to the authoritative values for the names of the NS RRset in".

File: draft-ietf-dnsop-ns-revalidation.mkd [hunk 2]
```diff
@@ -175,31 +177,33 @@ Attacks exploiting lack of this revalidation have been described in {{GHOST1}},
...
-    An additional reason is that there are unfortunately a number of nameservers in the field that (incorrectly) fail to properly answer explicit queries for zone apex NS records, and thus the revalidation logic may need to be applied lazily and opportunistically to deal with them.
-    In cases where the delegated nameservers respond incorrectly to an NS query, the resolver should abandon this algorithm for the zone in question and fall back to using only the information from the parent's referral response.
+The names from the NS RRset from a validating NS query may differ from the names in NS RRset in the referral response.
+Outstanding validation queries for "glue" that do not match names in the authoritative NS RRset be discarded, or they may be left running to completion.
+Their result will no longer be used in queries for the zone.
+Outstanding validation queries for "glue" that do match names in the authoritative NS RRset must be left running to completion.
+They do not need to be re-queried after reception of the authoritative NS RRset (see [](#upgrade-addresses)).
```
Comment (vixie) [inline draft-ietf-dnsop-ns-revalidation.mkd]: s/reception/receipt/

**Review by vixie — CHANGES_REQUESTED** [review]
some nits.

File: draft-ietf-dnsop-ns-revalidation.mkd (hunk 1 above)
Comment (wtoorop) [reply draft-ietf-dnsop-ns-revalidation.mkd]: Agree. Resolved in [e306cc0](https://github.com/shuque/ns-revalidation/pull/26/commits/e306cc0fc3d20f94c378c37c56282973320259bc)

File: draft-ietf-dnsop-ns-revalidation.mkd (hunk 2 above)
Comment (wtoorop) [reply draft-ietf-dnsop-ns-revalidation.mkd]: Thanks. Resolved in [a8303ea](https://github.com/shuque/ns-revalidation/pull/26/commits/a8303ea7b433fa5bf6dd462d36a576d19d926ec8)

**Pull request merged**